### PR TITLE
Update to Request and Requirement Errors

### PR DESCRIPTION
**Pull request by symorton**
**Problem:**
Now that we have a solid Error design, we need to update all external interfaces to adhere to that design.

**Solution:**
Updated the Request and Requirement classes to return VerifiedIdResult. Also updated the documentation.

**Validation:**
All unit tests pass.
Test cases in demo app work as expected.

**Type of change:**
- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [X] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [X] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: WalletLibrary/WalletLibrary/Errors/VerifiedIdErrors.swift [hunk 1]
```diff
@@ -9,21 +9,25 @@ enum VerifiedIdErrors {
...
         static let UnspecifiedError = "unspecified_error"
+        static let UserCanceled = "user_canceled"
     }
     
     /// Common Errors in Alphabetical Order.
-    case NetworkingError(message: String, correlationId: String, statusCode: String? = nil, innerError: Error? = nil)
-    case RequirementNotMet(message: String, correlationId: String? = nil)
+    case MalformedInput(error: Error, correlationId: String? = nil)
+    case NetworkingError(message: String, correlationId: String, statusCode: Int? = nil, innerError: Error? = nil)
+    case VCNetworkingError(error: NetworkingError)
```
Comment (logangirvin) [inline WalletLibrary/WalletLibrary/Errors/VerifiedIdErrors.swift:22]: Do we want to continue "VC" or should we move to "VID" ?

File: WalletLibrary/WalletLibrary/Utilities/VerifiedIdResult.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 public typealias VerifiedIdResult<T> = Result<T, VerifiedIdError>
 
-public extension VerifiedIdResult where Failure == VerifiedIdError {
+extension VerifiedIdResult where Failure == VerifiedIdError {
```
Comment (logangirvin) [inline WalletLibrary/WalletLibrary/Utilities/VerifiedIdResult.swift:8]: Does this not need to be public?

File: WalletLibrary/WalletLibraryTests/Errors/VerifiedIdErrorsTests.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        let mockVCNetworkingError = NetworkingError.badRequest(withBody: mockBody, statusCode: mockStatusCode)
+        
+        // Act
+        let result = VerifiedIdErrors.VCNetworkingError(error: mockVCNetworkingError).error as! VerifiedIdNetworkingError
+        
+        // Assert
+        XCTAssertEqual(result.statusCode, mockStatusCode)
+        XCTAssertEqual(result.message, "Bad Request")
+        XCTAssertEqual(result.code, VerifiedIdErrors.ErrorCode.NetworkingError)
+        if case NetworkingError.badRequest(let body, let statusCode) = (result.innerError as! NetworkingError) {
```
Comment (logangirvin) [inline WalletLibrary/WalletLibraryTests/Errors/VerifiedIdErrorsTests.swift:24]: should these be guard else fail?

File: README.md [hunk 4]
```diff
@@ -77,7 +77,7 @@ let validationResult = verifiedIdRequirement.validate()
 
 Once all of the requirements are fulfilled, you can double check that the request has been satisfied by calling the `isSatisfied` method on the request object. 
 ```Swift
-let isSatisfied = await presentationRequest.isSatisfied()
```
Comment (nithyaganeshng) [inline README.md:80]: await should be removed from all external interface calls in README, right?

Comment (symorton) [reply README.md:80]: This is a typo. isSatisfied should not have await but the rest should when some async process happens like network calls.

**Review by nithyaganeshng — APPROVED** [review]

**Review by logangirvin — APPROVED** [review]

**Pull request merged**